### ISSUE TITLE
DGX Nightly Benchmark run 20201214

### ISSUE THREAD

**Issue by quasiben** [quasiben/dask-scheduler-performance#55]
## Historical Throughput
<img width="641" alt="Benchmark Image"
src="https://raw.githubusercontent.com/quasiben/dask-scheduler-performance/benchmark-images/assets/dgx-20201214-benchmark-history.png">

## Raw Data
<Client: 'tcp://127.0.0.1:45981' processes=10 threads=10, memory=540.94 GB>
Distributed Version: 2020.12.0+5.g07970728
simple
	 5.612e-01 +/- 4.044e-02
shuffle
	 2.229e+01 +/- 9.666e-01
rand_access
	 1.037e-02 +/- 7.375e-04
anom_mean
	 1.008e+02 +/- 6.959e-01

## Raw Values
simple
	 [0.61208344 0.54716635 0.52957726 0.5543921  0.61655283 0.56562591
 0.59823322 0.55735683 0.47148418 0.55919671]
shuffle
	 [21.70394754 21.89360762 21.07760453 21.39884257 24.49116516 21.76838326
 22.0405066  22.54563832 22.63844275 23.39110851]
rand_access
	 [0.00853896 0.0112114  0.01039839 0.01051021 0.01061606 0.01033592
 0.01070213 0.01062369 0.01112819 0.0096302 ]
anom_mean
	 [100.05107999  99.66432261 100.75648022 100.26776743 101.68400311
 100.89423513 102.11114335 101.18314719 100.71105194 101.03717804]

## Dask Profiles

- [Shuffle
 Profile](https://raw.githack.com/quasiben/dask-scheduler-performance/benchmark-images/assets/20201214-shuffle-scheduler.html)
- [Random Access
  Profile](https://raw.githack.com/quasiben/dask-scheduler-performance/benchmark-images/assets/20201214-rand-access-scheduler.html)
- [Simple
  Profile](https://raw.githack.com/quasiben/dask-scheduler-performance/benchmark-images/assets/20201214-simple-scheduler.html)
- [Anom
  Mean](https://raw.githack.com/quasiben/dask-scheduler-performance/benchmark-images/assets/20201214-anom-mean-scheduler.html)

## Scheduler Execution Graph
<img width="641" alt="Sched Graph Image"
src="https://raw.githubusercontent.com/quasiben/dask-scheduler-performance/benchmark-images/assets/20201214-sched-graph.png">

**Comment by jakirkham** [comment]
The call graph here looks basically the same as what I'm seeing locally, which is good. That said, the percentages look significantly better than they did the benchmark the day before ( https://github.com/quasiben/dask-scheduler-performance/issues/54 ) despite no code changes to Distributed in the intervening time. Think we observed a similar issue last week. Am a bit curious about this. Is it possible Distributed is not getting updated?

**Comment by jakirkham** [comment]
Seems like we are already Cythonizing with `-f` (force), which would overwrite any existing C and recompile it.

https://github.com/quasiben/dask-scheduler-performance/blob/0bdbf29af38dc21abc4bca71f32221d9dea03fb7/nightly-benchmark/run-dgx-benchmark.sh#L24-L25

**Comment by quasiben** [comment]
Distributed is getting updated.  The raw data section lists `Distributed Version: 2020.12.0+5.g07970728`

**Comment by jakirkham** [comment]
Yeah I wonder if we are getting things out-of-order. Like running the shuffle benchmark and then updating Distributed as opposed to the other way around. This would explain the day lag we are seeing as it is up-to-date with the previous day's changes on the next day.

In any event I think the solution we came up with in PR ( https://github.com/quasiben/dask-cuda-benchmarks/pull/45 ) (and associated PRs) should fix that issue if that is indeed the cause. If not, it should give us more info for debugging and the process will be more streamlined, all of which should help 🙂
